### PR TITLE
media-gfx/gimp: add 2.99.6 development version

### PR DESCRIPTION
**Pull request by myrvogna**
[GIMP 2.99.6 development version announcement](https://www.gimp.org/news/2021/05/08/gimp-2-99-6-released/)
Dependencies' minimum versions updated, PYTHON_COMPAT updated, no major change needed.

Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Octiabrina Terrien-Puig <octiabrina@myrvogna.net>

### CODE REVIEW

**Comment by thesamesam** [comment]
@band-a-prend

**Comment by band-a-prend** [comment]
@myrvogna 
Thank you for created pull request.
Actually there are really no significant differences with configure.ac from 2.99.4.

In the 2.99.6 the build of gobject-introspection html documentation (docs for plugins development) was added. It requeres additional dependencies.

I added it to build under `doc` use flag together with common gtk-docs. The current problem is that there's no installation rule for this files and I want to install this documentation too. If I will do it soon then I could send changes here for rebase.

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( luajit )
+PYTHON_COMPAT=( python3_{7..10} )
+GNOME2_EAUTORECONF=yes
+VALA_MIN_API_VERSION="0.40"
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:9]: please, bump vala min version to portage tree available now:  
`VALA_MIN_API_VERSION="0.44"`

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+DESCRIPTION="GNU Image Manipulation Program"
+HOMEPAGE="https://www.gimp.org/"
+SRC_URI="mirror://gimp/v2.99/${P}.tar.bz2"
+LICENSE="GPL-3 LGPL-3"
+SLOT="0/3"
+KEYWORDS=""
+
+IUSE="aalib alsa aqua debug doc gnome heif javascript jpeg2k lua mng openexr postscript python udev unwind vala vector-icons webp wmf xpm cpu_flags_ppc_altivec cpu_flags_x86_mmx cpu_flags_x86_sse"
+REQUIRED_USE="lua? ( ${LUA_REQUIRED_USE} )
+	python? ( ${PYTHON_REQUIRED_USE} )"
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:23]:
```
REQUIRED_USE="
	lua? ( ${LUA_REQUIRED_USE} )
	python? ( ${PYTHON_REQUIRED_USE} )
"
```

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	>=dev-util/intltool-0.51.0
+	sys-apps/findutils
+	>=sys-devel/autoconf-2.54
+	>=sys-devel/automake-1.11
+	>=sys-devel/gettext-0.21
+	>=sys-devel/libtool-2.4.6
+	doc? (
+		>=dev-util/gtk-doc-1.32
+		dev-util/gtk-doc-am
+	)
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:106]: gimp 2.99.6 now provides gobject-introspection documentation for plugin developers:
```
	doc? (
		app-text/yelp-tools
		dev-libs/gobject-introspection[doctool]
		>=dev-util/gtk-doc-1.32
		dev-util/gtk-doc-am
	)
```

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+		--with-bug-report-url=https://bugs.gentoo.org/
+		--with-xmc
+		--without-libbacktrace
+		--without-webkit
+		--without-xvfb-run
+		$(use_enable cpu_flags_ppc_altivec altivec)
+		$(use_enable cpu_flags_x86_mmx mmx)
+		$(use_enable cpu_flags_x86_sse sse)
+		$(use_enable doc gtk_doc)
+		$(use_enable debug)
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:179]: `--enable-debug` as I see from gimp-2.99.x `configure.ac` only enables `DEBUGFLAG="-g"`.
As fas as I know the debugging information if required handled via setup https://wiki.gentoo.org/wiki/Knowledge_Base:Overriding_environment_variables_per_package

Therefore it and `IUSE="debug"` could be dropped - I forgot to drop last early.

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+		--with-appdata-test
+		--with-bug-report-url=https://bugs.gentoo.org/
+		--with-xmc
+		--without-libbacktrace
+		--without-webkit
+		--without-xvfb-run
+		$(use_enable cpu_flags_ppc_altivec altivec)
+		$(use_enable cpu_flags_x86_mmx mmx)
+		$(use_enable cpu_flags_x86_sse sse)
+		$(use_enable doc gtk_doc)
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:178]: Here we also enable gobject-introspection docs build:
```
		$(use_enable doc gtk_doc)
		$(use_enable doc g-ir-doc)
```

File: media-gfx/gimp/gimp-2.99.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+	# precedence on PDF documents by default
+	mv "${ED}"/usr/share/applications/{,zzz-}gimp.desktop || die
+
+	find "${D}" -name '*.la' -type f -delete || die
+
+	# Prevent dead symlink gimp-console.1 from downstream man page compression (bug #433527)
+	mv "${ED}"/usr/share/man/man1/gimp-console{-*,}.1 || die
+
+	_rename_plugins || die
+}
```
Comment (band-a-prend) [inline media-gfx/gimp/gimp-2.99.6.ebuild:247]: Unfortunately `makefile` currently doesn't provide `gobject-introspection docs` installation rule.
So the  manual installation is required. I hope the `gtk-doc` directory is OK for it.
```
	if use doc ; then
		mkdir "${ED}/usr/share/gtk-doc/html/gimp3_g-ir-docs" || die
		cp -r "${S}/devel-docs/g-ir-docs/html/"{gjs,python} "${ED}/usr/share/gtk-doc/html/gimp3_g-ir-docs/" || die
	fi
}
```

**Review by band-a-prend — CHANGES_REQUESTED** [review]
Please make some changes and then rebase the commit.

@thesamesam 
could you please comment additional changes offered here?

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 1 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild]: Wasn't sure about that one but it makes sense, done.

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 2 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild]: Done.

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 3 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild]: Done.

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 4 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild]: Done, both `use_enable debug` and `IUSE="debug"` dropped.

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 5 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild:182]: Done.

File: media-gfx/gimp/gimp-2.99.6.ebuild (hunk 6 above)
Comment (myrvogna) [reply media-gfx/gimp/gimp-2.99.6.ebuild:256]: Done.

**Comment by myrvogna** [comment]
@band-a-prend Thank you for the feedback, all the requested changes should be done now.

**Comment by band-a-prend** [comment]
Thank you.

Now we are waiting review/approve from @gentoo/proxy-maint team.

**Comment by thesamesam** [comment]
@band-a-prend thank you for reviewing this!
@myrvogna thanks for this!

Try coordinate together if you can on future bumps as @band-a-prend is pretty active. I'm sure you two can find some arrangement :)

Looks good!

**Pull request closed without merging**